### ISSUE TITLE
Visualization code of Figure 1 in paper.

### ISSUE THREAD

**Issue by MaureenZOU** [Atten4Vis/ConditionalDETR#5]
Hi Author, 

First thanks for your great work to improve the convergence speed of DETR with such a large margin. When reading the paper, I get a little bit confused on how do you exactly draw the attention map in Figure 1. 

Given object query q (1 x d), memory feature m (d x (hw)). I use the following equation to draw the attention maps:

Similarity(q,m) = Softmax(proj(q) \dot proj(m)) [1 x (hw)] where proj is the trained linear layer in cross attention module. 

The attention maps I get is quite similar with the one shown in DETR paper:

A random object query:
![Screen Shot 2021-09-13 at 10 46 41 AM](https://user-images.githubusercontent.com/11957155/133115426-34835c53-5dd2-42e1-bf1a-abef010d2972.png)

A random object query on head A:
![Screen Shot 2021-09-13 at 10 47 18 AM](https://user-images.githubusercontent.com/11957155/133115516-9fdd4919-7575-4510-aa86-ec66434ee24d.png)

A random object query on head B:
![Screen Shot 2021-09-13 at 10 47 25 AM](https://user-images.githubusercontent.com/11957155/133115539-74958d57-d98b-4fa9-9a17-3714b202c9bf.png)

A random object query on head C:
![Screen Shot 2021-09-13 at 10 47 30 AM](https://user-images.githubusercontent.com/11957155/133115568-9c68bbfd-9271-4757-875e-ca19fc3ac8ad.png)

Could you please give some information on how to generate attention in Figure 1? Thanks!

**Comment by SISTMrL** [comment]
hello, have you generated the attention map like fig. 1? @MaureenZOU

**Comment by MaureenZOU** [comment]
The problem was solved by the explanation in Section 3.4, paragraph comparison to detr. instead of measuring the similarity with memory + pos_encoding, the author just measuring the similarity between the position encoding.

**Comment by GWwangshuo** [comment]
> The problem was solved by the explanation in Section 3.4, paragraph comparison to detr. instead of measuring the similarity with memory + pos_encoding, the author just measuring the similarity between the position encoding.

Could you please kindly provide the souce code of visualization the attention map? That will be greatly helpful. Thanks a lot.

**Comment by GWwangshuo** [comment]
> The problem was solved by the explanation in Section 3.4, paragraph comparison to detr. instead of measuring the similarity with memory + pos_encoding, the author just measuring the similarity between the position encoding.

@MaureenZOU 
Could you please kindly provide the souce code of visualization the attention map? That will be greatly helpful. Thanks a lot.
